src/logic/calculate.js

```diff
@@ -0,0 +1,24 @@
+import operate from './operate';
+
+const calculate = ((data, buttonName) => {
+  let { total, next, operation } = data;
+
```

Comment:
Let's handle the case where `buttonName` is a number. The `calculate` function should be able to concatenate numbers